internal/resource/mutation/mutation.go

```diff
@@ -103,44 +114,47 @@ func apply(path *PathExpr, startIndex int, obj any, value any) error {
 	state := obj
 
 	for i, section := range path.ast.Sections[startIndex:] {
+		isLastSection := startIndex+i == len(path.ast.Sections)-1
+
 		// Map field indexing
-		if section.Field != nil {
+		if section.Field != nil || (section.Index != nil && section.Index.Key != nil) {
 			m, ok := state.(map[string]any)
 			if !ok {
 				continue
 			}
-			if startIndex+i == len(path.ast.Sections)-1 {
-				if value == nil {
-					delete(m, *section.Field)
-				} else {
-					m[*section.Field] = value
-				}
-				return nil
-			}
-			state = m[*section.Field]
-			continue
-		}
 
-		if section.Index == nil {
-			continue // should be impossible
-		}
-
-		// Alternative map field indexing
-		if key := section.Index.Key; key != nil {
-			m, ok := state.(map[string]any)
-			if !ok {
-				continue
+			var keyStr string
+			if section.Field != nil {
```

Comment:
When is this pre-populated vs not?

Reply:
I think the other comment about the logic addresses this?